lib/fishjam/component/hls/cluster/request_handler.ex

```diff
@@ -0,0 +1,39 @@
+defmodule Fishjam.Component.HLS.Cluster.RequestHandler do
+  @moduledoc """
+  Module responsible for handling HLS Retrieve Content requests in the cluster.
+  """
+
+  @behaviour Fishjam.Component.HLS.RequestHandler
+
+  alias Fishjam.{Room, RPCClient}
+
+  @local_module Fishjam.Component.HLS.Local.RequestHandler
+
+  @impl true
+  def handle_file_request(room_id, filename),
+    do: route_request(room_id, :handle_file_request, [room_id, filename])
+
+  @impl true
+  def handle_partial_request(room_id, filename),
+    do: route_request(room_id, :handle_partial_request, [room_id, filename])
+
+  @impl true
+  def handle_manifest_request(room_id, partial),
+    do: route_request(room_id, :handle_manifest_request, [room_id, partial])
+
+  @impl true
+  def handle_delta_manifest_request(room_id, partial),
+    do: route_request(room_id, :handle_delta_manifest_request, [room_id, partial])
+
+  defp route_request(room_id, fun, args) do
+    with {:ok, node} <- Room.ID.determine_node(room_id),
+         {:here?, false} <- {:here?, node == Node.self()},
```

Comment:
@Rados13 had a nice idea about this one. Instead of checking if node is self and then apply rpc or local mfa -> always use RPC even to self, so this one path in with will disappear.

Reply:
An awesome idea; in this case, though, we call different functions depending on whether the HLS content is on this node or another one -- so it can't really be done

Reply:
Yeah everywhere except `hls` I think this rule should apply :D

Reply:
Well then that's already implemented...